src/overlays/actors/ovl_En_Grasshopper/z_en_grasshopper.c

```diff
@@ -64,92 +138,990 @@ const ActorInit En_Grasshopper_InitVars = {
     (ActorFunc)EnGrasshopper_Draw,
 };
 
-// static ColliderJntSphElementInit sJntSphElementsInit[2] = {
-static ColliderJntSphElementInit D_809A8D1C[2] = {
+static ColliderJntSphElementInit sJntSphElementsInit[2] = {
     {
-        { ELEMTYPE_UNK3, { 0xF7CFFFFF, 0x00, 0x00 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_ON | TOUCH_SFX_NORMAL, BUMP_ON | BUMP_HOOKABLE, OCELEM_ON, },
-        { 7, { { 0, 0, 0 }, 0 }, 0 },
+        {
+            ELEMTYPE_UNK3,
+            { 0xF7CFFFFF, 0x00, 0x00 },
+            { 0xF7CFFFFF, 0x00, 0x00 },
+            TOUCH_ON | TOUCH_SFX_NORMAL,
+            BUMP_ON | BUMP_HOOKABLE,
+            OCELEM_ON,
+        },
+        { DRAGONFLY_LIMB_THORAX, { { 0, 0, 0 }, 0 }, 0 },
     },
     {
-        { ELEMTYPE_UNK2, { 0xF7CFFFFF, 0x07, 0x04 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_ON | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
-        { 6, { { 0, 0, 0 }, 0 }, 0 },
+        {
+            ELEMTYPE_UNK2,
+            { 0xF7CFFFFF, 0x07, 0x04 },
+            { 0xF7CFFFFF, 0x00, 0x00 },
+            TOUCH_ON | TOUCH_SFX_NORMAL,
+            BUMP_ON,
+            OCELEM_NONE,
+        },
+        { DRAGONFLY_LIMB_TAIL_TIP, { { 0, 0, 0 }, 0 }, 0 },
     },
 };
 
-// static ColliderJntSphInit sJntSphInit = {
-static ColliderJntSphInit D_809A8D64 = {
-    { COLTYPE_HIT2, AT_ON | AT_TYPE_ENEMY, AC_ON | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_1, OC2_TYPE_1, COLSHAPE_JNTSPH, },
-    ARRAY_COUNT(sJntSphElementsInit), D_809A8D1C, // sJntSphElementsInit,
+static ColliderJntSphInit sJntSphInit = {
+    {
+        COLTYPE_HIT2,
+        AT_ON | AT_TYPE_ENEMY,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_1,
+        OC2_TYPE_1,
+        COLSHAPE_JNTSPH,
+    },
+    ARRAY_COUNT(sJntSphElementsInit),
+    sJntSphElementsInit,
 };
 
-#endif
+typedef enum {
+    /* 0 */ EN_GRASSHOPPER_ANIM_RAISE_TAIL,
+    /* 1 */ EN_GRASSHOPPER_ANIM_LOWER_TAIL,
+    /* 2 */ EN_GRASSHOPPER_ANIM_FLY,
+    /* 3 */ EN_GRASSHOPPER_ANIM_ATTACK,
+    /* 4 */ EN_GRASSHOPPER_ANIM_HOVER,
+    /* 5 */ EN_GRASSHOPPER_ANIM_DAMAGE,
+    /* 6 */ EN_GRASSHOPPER_ANIM_DEAD,
+    /* 7 */ EN_GRASSHOPPER_ANIM_FALL,
+} EnGrasshopperAnim;
+
+static AnimationHeader* sAnimations[] = {
+    &gDragonflyRaiseTailAnim, &gDragonflyLowerTailAnim, &gDragonflyFlyAnim,  &gDragonflyAttackAnim,
+    &gDragonflyHoverAnim,     &gDragonflyDamageAnim,    &gDragonflyDeadAnim, &gDragonflyFallAnim,
+};
+
+static u8 sAnimationModes[] = {
+    ANIMMODE_ONCE, ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_ONCE,
+    ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_ONCE, ANIMMODE_ONCE,
+};
```

Comment:
This looks like it and the enum would go better above (or in?) the animation-changing function. Might also want to put enum comments in, at least on the modes.